Proposals: Improvements in Review Process

ananyo2012 opened the pull request:
Change form label and remove vote values from ProposalReviewerVote form

Add filter for Proposal Section Reviwer

Make radio buttons horizontally aligned in the Vote form

Use nickname for showing votes and remove vote value from Reviwer dashboard

Fixes #560

jaidevd commented:
Sure.

jaidevd commented:
Please note that 'sum of votes' is not what we need as shown here: https://user-images.githubusercontent.com/10486343/42236449-7e521fe8-7f18-11e8-8060-6c52056b470a.png, 

but we need the actual individual votes. Please treat the votes as Enums, not as integer values.

Thanks

ananyo2012 commented on junction/proposals/forms.py:
```diff
@@ -53,8 +53,7 @@ def _get_proposal_section_reviewer_vote_choices(conference):
             continue
-        values.append((i.vote_value, '{} ({})'.format(
-            i.description, i.vote_value)))
+        values.append((i.vote_value, '{}'.format(i.description)))
```
Compare [first screenshot](https://user-images.githubusercontent.com/10486343/42195050-a91aa706-7e94-11e8-83b1-e63accb16dd1.png) vs [second screenshot](https://user-images.githubusercontent.com/10486343/42236443-796b07b0-7f18-11e8-88bd-0fe9ce425a3a.png). To remove the vote values from the form.

ananyo2012 commented:
> Please note that 'sum of votes' is not what we need as shown here: https://user-images.githubusercontent.com/10486343/42236449-7e521fe8-7f18-11e8-8060-6c52056b470a.png,

but we need the actual individual votes. Please treat the votes as Enums, not as integer values.

Didn't get your point clearly. Do I need to keep the individual vote values in the display and remove the total sum of votes ? About enums I will make that change.

ananyo2012 commented:
@jaidevd Yes the data will be extractable by any client. The changes I made are mostly for the fronted changes. I didn't change any model or view methods.

jaidevd commented:
@ananyo2012 ,

> Do I need to keep the individual vote values in the display and remove the total sum of votes?

Yes, we do not want to sum the vote values at all, anywhere.

> The changes I made are mostly for the fronted changes. I didn't change any model or view methods.

Out of curiosity, if no model changes are made, then how will we be able to store the individual vote values? Because IIRC, the choices last time were different, no?

Thanks

ananyo2012 commented:
> Yes, we do not want to sum the vote values at all, anywhere.

Ok I won't display the sum of total votes. But do I need to display the value of individual votes in brackets ?

> Out of curiosity, if no model changes are made, then how will we be able to store the individual vote values? Because IIRC, the choices last time were different, no?

I just changed the entries in the database. There are 2 tables one for `Vote Values` (Where there are fields for description and Vote values) and other for `Reviewer Votes` which has fields for Proposal, Voter, Vote Value etc. I just updated the entries in `Vote Values` database.

jaidevd commented:
> do I need to display the value of individual votes in brackets?
No.
Cool, then. Feel free to merge.

ananyo2012 commented on junction/proposals/templatetags/proposal_filters.py:
```diff
@@ -53,21 +53,26 @@ def has_upvoted_comment(comment, user):
 
 
+@register.filter(name='proposal_section_reviewer')
+def proposal_section_reviewer(proposal, user):
```
Fixed
